tests/Test.hs

```diff
@@ -148,5 +318,6 @@ main = defaultMain $ testGroup "Tests"
     , testCase "groupBy: Char order" groupByCharOrder
     , testCase "findIndexOrEnd" goodFindIndex
     , testCase "Stream Interop" firstI
+    , testCase "readInt" readIntCases
```

Comment:
Do you have a coverage report? It's difficult to check otherwise that all branches are tested.

Reply:
Definitely not - the library overall is somewhat under-tested.

Reply:
I don't have a coverage report. Any pointers on how to get one?

Reply:
`stack test --coverage` and probably a similar incantation with Cabal.
`stack` is somewhat fragile in this aspect, you may have a better mileage with [`stack-2.5.0.1-rc`](https://github.com/commercialhaskell/stack/releases/tag/v2.5.0.1)

Reply:
In terms of coverage, I made sure that overflow does not happen at maxBound/minBound, and does happen directly outside that range.  I also made sure that we're not performing the wrong number of I/O ops by rerunning already performed effects. And testing that leading zeros are handled correctly, and chunk boundaries are not an issue just after a sign, etc. and that the unconsumed tail of the string is correct.  Also tested that just too many leading zeros are rejected, and that overflow reconstruction is returning the original input.

It would be nice to know which branches are not hit if any, what tools expose that?

Reply:
```
cabal test --enable-coverage
```
or 
```
stack upgrade --binary-version 2.5.0.1
stack test --coverage
```
Both options will produce a detailed HTML report, hightlighting not hit branches.

Reply:
Wow!  That was painless.  Got the code coverage to 100% (for the new code in this PR only) in just ~20 minutes. Just needed to add a couple of simple test cases, and replace a couple of variables with their constant value in that context. There is a cosmetic never unevaluated `0` and `()`, but those are [obviously fine](http://imrryr.org/~viktor/sbs-hpc/test/streaming-bytestring-0.1.6-inplace/Data.ByteString.Streaming.Char8.hs.html)